src/Kafka/Consumer.hs

```diff
@@ -113,17 +113,20 @@ newConsumer :: MonadIO m
             => ConsumerProperties
             -> Subscription
             -> m (Either KafkaError KafkaConsumer)
-newConsumer props (Subscription ts tp) = liftIO $ do
-  let cp = setCallback (rebalanceCallback (\_ _ -> return ())) <> props
+newConsumer props@ConsumerProperties { cpUserPolls = Any cup } (Subscription ts tp) = liftIO $ do
+  let cp = case cup of
+        False -> setCallback (rebalanceCallback (\_ _ -> return ())) <> props
+        True  -> props
```

Comment:
@shlevy Why do we ignore rebalance callback for synchronous polls?

Reply:
We don't ignore any user-supplied rebalance callback, but if there is no user-supplied one then we just let rdkafka's default impl take over.

Reply:
@shlevy Ah, I see, thanks!